calyx-py/calyx/queue_data_gen.py

```diff
@@ -56,12 +56,19 @@ def no_err_cmds_list(queue_size, num_cmds):
     return commands
 
 
-def dump_json(num_cmds, no_err: bool, queue_size: Optional[int] = None):
+def dump_json(num_cmds, no_err: bool, queue_size: Optional[int] = None, nwc=False):
     """Prints a JSON representation of the data to stdout.
     The data itself is populated randomly, following certain rules:
     - It has three "memories": `commands`, `values`, and `ans_mem`.
-    - The `commands` memory has `num_cmds` items, which are 0, 1, or 2.
-      0: pop, 1: peek, 2: push
+    - Optional memories `ranks` and `times` are included for queues primed for non-work-conserving algorithms.
+    - The `commands` memory has `num_cmds` items, which range from 0-4. They are as follows:
+
+        0 : pop (for non-work-conserving algorithms, pop by predicate)
```

Comment:
This docstring has a bug, I think? 0, 1, 3, and 4 are all for NWC? Also, please explain that passing the always-true predicate will allow predicate-having `pop` and `peek` to behave as though they do not have predicates.

Reply:
I don't know if I have the `always-true` functionality set up, but can work on it.

Reply:
Set up now, and documented!